examples/snap-plugin-collector-rand/rand/rand.go

```diff
@@ -54,8 +55,23 @@ var (
 	}
 )
 
+var req bool = false
+
 func init() {
 	rand.Seed(42)
+
+	//The required-config flag is added for testing plugin-lib-go.
+	//When the flag is set, an additional policy will be added in GetConfigPolicy().
+	//This additional policy has a required field. This simulates
+	//the situation when a plugin requires a config to load.
+	plugin.AddFlag(
+		cli.BoolFlag{
+			Name:        "required-config",
```

Comment:
I think that providing the hook for plugin authors to use flags has value especially when considering stand-alone plugins.  For instance, an author of a standalone plugin may want to expose custom flags.  I was thinking that when we update plugin authoring we would highlight how authors can add and use flags.